CraftTweaker2-API/src/main/java/crafttweaker/api/event/ProjectileImpactArrowEvent.java

```diff
@@ -0,0 +1,48 @@
+package crafttweaker.api.event;
+
+import crafttweaker.annotations.ZenRegister;
+import crafttweaker.api.entity.IEntity;
+import crafttweaker.api.entity.IEntityLivingBase;
+import crafttweaker.api.item.IItemStack;
+import crafttweaker.api.world.IRayTraceResult;
+import stanhebben.zenscript.annotations.ZenClass;
+import stanhebben.zenscript.annotations.ZenGetter;
+import stanhebben.zenscript.annotations.ZenMethod;
+import stanhebben.zenscript.annotations.ZenSetter;
+
+@ZenClass("crafttweaker.event.ProjectileImpactArrowEvent")
+@ZenRegister
+public interface ProjectileImpactArrowEvent extends ProjectileImpactEvent, IEventCancelable {
+  @ZenGetter("arrow")
+  IEntity getArrow();
+
+  @ZenGetter("shooter")
+  IEntity getShooter();
+
+  @ZenGetter("damage")
+  double getDamage();
+
+  @ZenSetter("damage")
+  void setDamage (double damage);
+
+  @ZenSetter("knockbackStrength")
+  void setKnockbackStrength (int knockbackStrength);
```

Comment:
Just to clarify:
Do they also have a way to get that value, or only a way to set it?

Reply:
These methods are short-hand accessors into the actual EntityArrow to avoid writing an interface for EntityArrow. As far as I'm aware, `knockbackStrength` for EntityArrow is a private int, and it only has a setter; I decided to not implement a getter, but didn't want to short-change people on things that they would otherwise be able to access.